### ISSUE TITLE
SetKeyboardFocusHere regression (?) from 1.83 to 1.88.

### ISSUE THREAD

**Issue by slotthhy** [ocornut/imgui#5457]
**Version/Branch of Dear ImGui:**

1.83 from 1b435ae3e07ca813eb3ef40aaabe7053f5570fae
```
Dear ImGui 1.83 (18300)
--------------------------------
sizeof(size_t): 8, sizeof(ImDrawIdx): 2, sizeof(ImDrawVert): 20
define: __cplusplus=199711
define: _WIN32
define: _WIN64
define: _MSC_VER=1932
define: _MSVC_LANG=201402
define: IMGUI_HAS_VIEWPORT
define: IMGUI_HAS_DOCK
--------------------------------
io.BackendPlatformName: imgui_impl_win32
io.BackendRendererName: imgui_impl_dx11
io.ConfigFlags: 0x00000441
 NavEnableKeyboard
 DockingEnable
 ViewportsEnable
io.ConfigViewportsNoDecoration
io.ConfigInputTextCursorBlink
io.ConfigWindowsResizeFromEdges
io.ConfigMemoryCompactTimer = 60.0
io.BackendFlags: 0x00001C0E
 HasMouseCursors
 HasSetMousePos
 PlatformHasViewports
 HasMouseHoveredViewport
 RendererHasVtxOffset
 RendererHasViewports
--------------------------------
io.Fonts: 1 fonts, Flags: 0x00000000, TexSize: 512,64
io.DisplaySize: 1264.00,761.00
io.DisplayFramebufferScale: 1.00,1.00
--------------------------------
style.WindowPadding: 8.00,8.00
style.WindowBorderSize: 1.00
style.FramePadding: 4.00,3.00
style.FrameRounding: 0.00
style.FrameBorderSize: 0.00
style.ItemSpacing: 8.00,4.00
style.ItemInnerSpacing: 4.00,4.00
```

1.88 from 9cd9c2eff99877a3f10a7f9c2a3a5b9c15ea36c6
```
Dear ImGui 1.88 (18800)
--------------------------------
sizeof(size_t): 8, sizeof(ImDrawIdx): 2, sizeof(ImDrawVert): 20
define: __cplusplus=199711
define: _WIN32
define: _WIN64
define: _MSC_VER=1932
define: _MSVC_LANG=201402
define: IMGUI_HAS_VIEWPORT
define: IMGUI_HAS_DOCK
--------------------------------
io.BackendPlatformName: imgui_impl_win32
io.BackendRendererName: imgui_impl_dx11
io.ConfigFlags: 0x00000441
 NavEnableKeyboard
 DockingEnable
 ViewportsEnable
io.ConfigViewportsNoDecoration
io.ConfigInputTextCursorBlink
io.ConfigWindowsResizeFromEdges
io.ConfigMemoryCompactTimer = 60.0
io.BackendFlags: 0x00001C0E
 HasMouseCursors
 HasSetMousePos
 PlatformHasViewports
 HasMouseHoveredViewport
 RendererHasVtxOffset
 RendererHasViewports
--------------------------------
io.Fonts: 1 fonts, Flags: 0x00000000, TexSize: 512,64
io.DisplaySize: 1264.00,761.00
io.DisplayFramebufferScale: 1.00,1.00
--------------------------------
style.WindowPadding: 8.00,8.00
style.WindowBorderSize: 1.00
style.FramePadding: 4.00,3.00
style.FrameRounding: 0.00
style.FrameBorderSize: 0.00
style.ItemSpacing: 8.00,4.00
style.ItemInnerSpacing: 4.00,4.00
```

**My Issue/Question:**

It seems there is a regression in SetKeyboardFocusHere. In 1.83, you can see below that no matter what or where is clicked, the text box is focused as long as the window is focused. In 1.88 this is not the case. Not sure if this is a regression or intended, but I appreciate anyone taking a look.

**Screenshots/Video**
https://user-images.githubusercontent.com/6005409/178127087-7544e52e-723e-47c7-a71d-f2ca0e88f3fe.mp4

**Standalone, minimal, complete and verifiable example:** _(see https://github.com/ocornut/imgui/issues/2261)_
```
ImGui::Begin("Hello, world!");                          // Create a window called "Hello, world!" and append into it.

static char buf[240];

//ImGui::InputTextWithHint("###test", "Search...", buf, 240, ImGuiInputTextFlags_NoUndoRedo);
ImGui::InputText("###test", buf, 240, ImGuiInputTextFlags_NoUndoRedo);

if (ImGui::IsWindowFocused(ImGuiFocusedFlags_RootAndChildWindows) && !ImGui::IsAnyItemActive() && !ImGui::IsMouseClicked(ImGuiMouseButton_Left))
    ImGui::SetKeyboardFocusHere(0);

ImGui::Text("Application average %.3f ms/frame (%.1f FPS)", 1000.0f / ImGui::GetIO().Framerate, ImGui::GetIO().Framerate);
ImGui::End();
```
The above code will produce different behavior across 1.83 and 1.88, and I don't believe there was any documentation that this was breaking.

**Comment by ocornut** [comment]
The correct code is calling SetKeyboardFocusHere(-1). Using 0 would pick the next item.

I think your code was relying on a bug/undefined behavior where the value would wrap around because that code used the same logic as Tabbing which does wrap around. But we disabled the wrapping in that API.

**Comment by lmcintyre** [comment]
Sounds good. I had assumed this was unintended behavior on 1.83's end, which was fixed in 1.88 but I wanted to make sure. Thanks.
